modern/src/reports/ReportFilter.js

```diff
@@ -59,17 +69,17 @@ const ReportFilter = ({ children, handleSubmit, showOnly }) => {
 
   return (
     <Grid container spacing={2} justify="flex-end">
-      <Grid item xs={12} sm={period === 'custom' ? 3 : 6}>
+      <Grid item xs={fullScreen ? 6 : 12} sm={!fullScreen && period === 'custom' ? 3 : 6}>
         <FormControl variant="filled" fullWidth>
           <InputLabel>{t('reportDevice')}</InputLabel>
-          <Select value={deviceId} onChange={(e) => setDeviceId(e.target.value)}>
+          <Select value={deviceId} onChange={handleDeviceChange}>
             {devices.map((device) => (
               <MenuItem value={device.id}>{device.name}</MenuItem>
             ))}
           </Select>
         </FormControl>
       </Grid>
-      <Grid item xs={12} sm={period === 'custom' ? 3 : 6}>
+      <Grid item xs={fullScreen ? 6 : 12} sm={!fullScreen && period === 'custom' ? 3 : 6}>
```

Comment:
Is there some way to check `fullScreen` on the parent instead instead of doing to every single item?